lib/Experian/IDAuth.pm

```diff
@@ -42,27 +42,26 @@ sub set {
 
 sub get_result {
     my $self = shift;
-    $self->_do_192_authentication || return;
-    for ($self->{search_option}) {
-        /ProveID_KYC/ && return $self->_get_result_proveid;
-        /CheckID/     && return $self->_get_result_checkid;
-        die "invalid search_option $_";
-    }
-    return;
+
+    my $search_option = $self->{search_option};
+
+    die "invalid search_option $search_option" if $search_option !~ /(?:ProveID_KYC|CheckID)/;
```

Comment:
Since this is a module for public use, would `Carp::croak()` be more appropriate?